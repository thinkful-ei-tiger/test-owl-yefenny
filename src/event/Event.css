.event{
    color: darkgray;
    margin: 15px 20px;
}

.message-type {
    display: flex;
   
}

.message-avatar img{
    width: 50px;
    border-radius: 50%;
}

.message-information{
    display: flex;
    flex-direction: column;
}

.message-header{
    display: flex;
    justify-content: center;
    margin-left: 5px;
}

.message-header h2{
   font-size: 16px;
   margin-bottom: 0;
   color: gray;
}

.message-header span {
    margin: 13px 0 0 5px;
}