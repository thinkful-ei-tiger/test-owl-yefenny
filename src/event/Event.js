import React, { Component } from 'react';
import PARTICIPANTS from '../dummy';
import { format } from 'date-fns';
import './Event.css';

export default class Event extends Component {
  getParticipant(id) {
    const participant = PARTICIPANTS.find(
      (participant) => participant.id === id
    );
    return participant;
  }
  handleType(event) {
    const participant = this.getParticipant(event.participantId);

    switch (event.type) {
      case 'message':
        return (
          <div className='message-type'>
            <div className='message-avatar'>
              <img src={participant.avatar} alt='profile' />
            </div>
            <div className='message-information'>
              <div className='message-header'>
                <h2>{participant.name}</h2>
                <span>{format(new Date(event.time), 'KK:m aaaa')}</span>
              </div>
              <div className='message'>
                <p>{event.message}</p>
              </div>
            </div>
          </div>
        );
        break;

      case 'thumbs-up':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> gave a thumbs up 👍
            </span>
          </div>
        );
        break;
      case 'thumbs-down':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> gave a thumbs down 👎
            </span>
          </div>
        );
        break;
      case 'raise-hand':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> raise their hand ✋
            </span>
          </div>
        );
        break;
      case 'clap':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> clapped 👏
            </span>
          </div>
        );
        break;
      case 'join':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> joined
            </span>
          </div>
        );
        break;
      case 'leave':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> left
            </span>
          </div>
        );
        break;
      case 'join-stage':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> joined the stage
            </span>
          </div>
        );
        break;
      case 'leave-stage':
        return (
          <div>
            <span>
              <strong>{participant.name}</strong> left the stage
            </span>
          </div>
        );
        break;
    }
  }
  render() {
    const { event } = this.props;
    const content = this.handleType(event);
    return <div className='event'>{content}</div>;
  }
}
