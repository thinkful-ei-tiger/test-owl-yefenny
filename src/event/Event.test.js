import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Event from './Event';

describe('Event Component', () => {
  const event = {
    participantId: 3,
    type: 'message',
    message: 'Hello world',
    time: 1548852646559,
    timestamp: 1548852484247
  };
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Event event={event} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('UI renders without problem', () => {
    const tree = renderer.create(<Event event={event} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
