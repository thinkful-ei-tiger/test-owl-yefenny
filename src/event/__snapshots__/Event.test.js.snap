// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component UI renders without problem 1`] = `
<div
  className="event"
>
  <div
    className="message-type"
  >
    <div
      className="message-avatar"
    >
      <img
        alt="profile"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="message-information"
    >
      <div
        className="message-header"
      >
        <h2>
          Axe Kubicka
        </h2>
        <span>
          07:50 a.m.
        </span>
      </div>
      <div
        className="message"
      >
        <p>
          Hello world
        </p>
      </div>
    </div>
  </div>
</div>
`;
